bin/cli.js

```diff
@@ -187,36 +199,53 @@ const getConfig = () => {
   program.version(version).usage('[options] -- <file ...>');
   cliOptions.forEach(option => program.option(...option));
   program.parse(process.argv);
+  if (program.quiet && program.verbose) {
+    console.error('Quiet and verbose options are mutually exclusive');
+    process.exit(1);
+  }
 
   const config = program.config ? parseFile(program.config) : {};
   getEnvironment(config, program);
 
   config.files = merge(config.files, program.args);
   config.files = loadFiles(config.files);
-
   config.exclude = merge(config.exclude, program.exclude);
   config.files = exclude(config.files, config.exclude);
   config.nodeOnly = merge(config.nodeOnly, program.nodeOnly);
   config.browserOnly = merge(config.browserOnly, program.browserOnly);
+  if (program.quiet) {
+    config.verbosity = 0;
+  } else if (program.verbose > logLevels.length) {
+    config.verbosity = logLevels.length;
+  } else {
+    config.verbosity = program.verbose || 1;
+  }
 
   if (config.environments.includes('browser')) {
     config.browser = config.browser || {};
-
-    config.browser.browsers = merge(
-      config.browser.browsers, program.browsers);
-    config.browser.logLevel = program.browserLog || config.browser.logLevel;
+    config.browser.browsers = merge(config.browser.browsers, program.browsers);
+    if (program.browserLog) {
+      config.browser.logLevel = program.browserLog;
+    } else if (!isLogAtLeast(config.verbosity, config.browser.logLevel)) {
+      config.browser.logLevel = logLevels[config.verbosity];
+    }
     config.browser.port = +program.browserPort || config.browser.port;
     config.browser = getBrowserConfig(config);
   }
   return config;
 };
 
 const runNode = (config, cb) => {
+  if (config.verbosity === 0) metatests.runner.instance.removeReporter();
   metatests.runner.instance.on('finish', () => {
-    console.log('Tests finished. Waiting for unfinished tests after end...\n');
+    if (isLogAtLeast(config.verbosity, 'error')) {
```

Comment:
Why is that printed on 'error' logging level? Isn't it supposed to be used to only log errors?

Reply:
There should be output at all levels except `disable` which represents quiet mode. And on log level we will be outputting config. And since `metatests` will be splitted into `metatests` for node and for browser, those lines will be deleted anyway.

Reply:
About removed/obsolete: Every PR is supposed to be at least somewhat self-complete/self-contained, so either you remove those lines here right away (as they will be removed later) or you should put them under proper log level because we may release a new version after this PR and it would not be correct to print those under `error` level.

I do think this should be printed always unless level is `disabled` though it's hard to associate log-levels to our Reporters as basically, we should always print at least the resulting test-stats (passed/failed) though that already doesn't fall under `error` category and in the least goes into `info` which is not the default level. And even if we make `info` the default then it'd be strange to have other `error`, `warn` levels because `-vv` is supposed to be counted from the default. That's why whole `verboseness` idea doesn't really fit under Test Framework because it is useless to have testing framework that doesn't print anything (usually).

I think we should at most leave `disable`, `default` and `debug` error levels and have `--show-config` as a separate option if we need it and add other `--reporter` option that will accept which class to use as a reporter.
Also, it was initially my idea to add `--verboseness` but now I think that it doesn't really fit in and we shuold remove that.
/cc @SemenchenkoVitaliy @belochub @nechaido

Reply:
I'm ok with bringing back `--show-config`, but I think we should just leave `--quiet` instead of `--verboseness` or `--reporter`. `metatests` in node doesn't have different log levels, so all we need is to print results of testing or not to print them. As for browser - it has its own separate option for log levels.

Reply:
I think it'll be better to just output config on `debug` logging level, instead of adding another `--show-config` option.

Reply:
`--reporter` doesn't correlate with current `veboseness` (or other) options. It should be there to be able to define how you want your results.

I'm ok with removing verboseness completely in favor of only `quiet`+`show-config` option
OR
having 'disable', `default`, `debug` and then remove `show-config` and print it in `debug` mode.

Reply:
As for me, I like the first option more because otherwise the only difference between `default` and `debug` will be whether to display config or not. Not including browser log levels which will be limited but not determined by command line options.